src/test/java/seedu/address/model/poc/UniquePocListTest.java

```diff
@@ -0,0 +1,169 @@
+package seedu.address.model.poc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_HUSBAND;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalPoc.ALICE;
+import static seedu.address.testutil.TypicalPoc.BOB;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.poc.exceptions.DuplicatePocException;
+import seedu.address.model.poc.exceptions.PocNotFoundException;
+import seedu.address.testutil.PocBuilder;
+
+public class UniquePocListTest {
+
+    private final UniquePocList uniquePocList = new UniquePocList();
+
+    @Test
+    public void contains_nullPoc_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniquePocList.contains(null));
+    }
+
+    @Test
+    public void contains_companyNotInList_returnsFalse() {
+        assertFalse(uniquePocList.contains(ALICE));
+    }
+
+    @Test
+    public void contains_companyInList_returnsTrue() {
+        uniquePocList.add(ALICE);
+        assertTrue(uniquePocList.contains(ALICE));
+    }
+
+    @Test
+    public void contains_companyWithSameIdentityFieldsInList_returnsTrue() {
+        uniquePocList.add(ALICE);
+        Poc editedAlice = new PocBuilder(ALICE).withTags(VALID_TAG_HUSBAND)
```

Comment:
This tag may not be appropriate in the context of a POC of a company

Reply:
Perhaps we can all work on this together in the next iteration as there are a lot of tags to change.